src/library_pthread.js

```diff
@@ -90,6 +90,7 @@ var LibraryPThread = {
       PThread['setThreadStatus'] = PThread.setThreadStatus;
       PThread['threadCancel'] = PThread.threadCancel;
       PThread['threadExit'] = PThread.threadExit;
+      PThread['setExitStatus'] = PThread.setExitStatus;
```

Comment:
why is this in this PR?

Reply:
We need to preserve the exit code of each thread so that that then it does become joinable the joining thread sees the correct exit value.   

This happens when thread's main functions returns N, but the thread is kept alive for some reason.  Once that reason is not longer there, the thread is joinable, and we still want the join'ing thread to get N as the result.